routes/auth.js

```diff
@@ -5,22 +5,27 @@ const config = require('../config');
 const {logout} = require('../controllers/auth-api')
 const {storeLoginEvent, storeLogoutEvent} = require("../neo4j/neo4j-operations");
 const {formatVariables, formatMap} = require("../bento-event-logging/const/format-constants");
+const {createToken, verifyToken} = require("../services/tokenizer");
 
 /* Login */
-router.post('/login', async function (req, res) {
+/* Granting an authenticated token */
+router.post(['/login', '/token-login'], async function (req, res) {
     try {
         const reqIDP = config.getIdpOrDefault(req.body['IDP']);
         const { name, lastName, tokens, email, idp } = await idpClient.login(req.body['code'], reqIDP, config.getUrlOrDefault(reqIDP, req.body['redirectUri']));
-        req.session.userInfo = {
+        const userInfo = formatVariables({
             email: email,
             IDP: idp,
             firstName: name,
             lastName: lastName
-        };
-        req.session.userInfo = formatVariables(req.session.userInfo, ["IDP"], formatMap);
-        await storeLoginEvent(req.session.userInfo.email, req.session.userInfo.IDP);
-        req.session.tokens = tokens;
-        res.json({name, email, "timeout": config.session_timeout / 1000});
+        }, ["IDP"], formatMap);
+
+        if (!req.originalUrl.includes("/token-login")) {
+            req.session.userInfo = userInfo
+            req.session.tokens = tokens;
+        }
+        await storeLoginEvent(userInfo.email, userInfo.IDP);
+        res.json({name, email, accessToken: await createToken(userInfo), "timeout": config.session_timeout / 1000});
```

Comment:
token-login response is returning the session_timeout instead of the token_timeout